social-network-api/src/main/java/fs/socialnetworkapi/service/LikeService.java

```diff
@@ -1,35 +1,48 @@
 package fs.socialnetworkapi.service;
 
+import fs.socialnetworkapi.dto.post.PostDtoOut;
 import fs.socialnetworkapi.entity.Like;
 import fs.socialnetworkapi.entity.Post;
 import fs.socialnetworkapi.entity.User;
 import fs.socialnetworkapi.exception.PostNotFoundException;
-import fs.socialnetworkapi.exception.UserNotFoundException;
 import fs.socialnetworkapi.repos.LikeRepo;
 import fs.socialnetworkapi.repos.PostRepo;
-import fs.socialnetworkapi.repos.UserRepo;
 import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.Optional;
 
 @Service
 @AllArgsConstructor
 public class LikeService {
 
   private final LikeRepo likeRepo;
-  private final UserRepo userRepo;
   private final PostRepo postRepo;
+  private final ModelMapper mapper;
+  private final User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
 
-  public List<Like> getLikesForUser(Long userId) {
```

Comment:
`private final User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();`
need to be obtained in each method

Reply:
Ok. Now I'm using private User getUser() method for it. No reason for changing all methods when/if logic of receiving user would be changed.